src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentMethodsScreen.tsx

```diff
@@ -235,6 +250,58 @@ const GenericError = () => {
   );
 };
 
+export const useOnRecurringPaymentReceived = ({
+  recurringPaymentId,
+  callback,
+}: {
+  recurringPaymentId: number | undefined;
+  callback: () => void;
+}) => {
+  const {userId} = useAuthContext();
+
+  const mapRecurringPayment = (
+    recurringPayment: FirebaseFirestoreTypes.DocumentData,
+  ): RecurringPayment => {
+    return {
+      id: recurringPayment.id,
+      expires_at: recurringPayment.expires_at,
+      masked_pan: recurringPayment.masked_pan,
+      payment_type: recurringPayment.payment_type,
+    };
+  };
+
+  useEffect(() => {
+    const recurringPaymentsUnsub = firestore()
+      .collection('customers')
+      .doc(userId)
+      .collection('recurringPayments')
+      .orderBy('created', 'desc')
+      .onSnapshot(
+        (snapshot) => {
+          const recurringPayments =
+            snapshot.docs.map<RecurringPayment>(mapRecurringPayment);
+          if (
+            recurringPayments.some(
+              (rp) =>
+                rp.id === recurringPaymentId &&
+                recurringPaymentId !== undefined,
```

Comment:
When `recurringPaymentId` is undefined we don't need this subscription at all? As of now it subscribes and iterates through all elements just to always evaluate to `false` for every item and not run the `callback`.

Rather check whether `recurringPaymentId` is defined at the top of the effect and do an early return if it is not.